src/contrib/cluster/Akka.Cluster.Sharding/PersistentShard.cs

```diff
@@ -45,29 +93,39 @@ public class PersistentShard : Shard
         /// <param name="extractShardId">TBD</param>
         /// <param name="handOffStopMessage">TBD</param>
         public PersistentShard(
+            IActorContext context,
+            IPersistentActorContext persistentContext,
+            Action<object> unhandled,
```

Comment:
Yes, but it breaks code compatibility between .net and java Akka, making future code port harder. And even if it is internal, someone might be using it since it is public.